go-controller/pkg/ovn/ovn.go

```diff
@@ -629,6 +658,28 @@ func (oc *Controller) WatchNodes() error {
 				klog.Warningf(err.Error())
 				gatewaysFailed.Store(node.Name, true)
 			}
+
+			//add any existing egressFirewall objects to join switch
+			namespaceList, err := oc.watchFactory.GetNamespaces()
+			if err != nil {
+				klog.Errorf("Error getting list of namespaces when adding node: %s", node.Name)
+			}
+			for _, namespace := range namespaceList {
+				nsInfo, err := oc.waitForNamespaceLocked(namespace.Name)
+				if err != nil {
+					klog.Errorf("Failed to wait for namespace %s event (%v)",
+						namespace.Name, err)
+					continue
+				}
+				if nsInfo.egressFirewallPolicy != nil {
+					err = nsInfo.egressFirewallPolicy.addACLToJoinSwitch(joinSwitch(node.Name), nsInfo.addressSet.GetIPv4HashName())
```

Comment:
You'll want to check IPv4 / IPv6 here, right now it's just using `GetIPv4HashName`

Reply:
re organized the code and and checked the IP version. Done